aten/src/ATen/core/op_registration/test_helpers.h

```diff
@@ -79,3 +79,9 @@ void expectListEquals(c10::ArrayRef<T> expected, std::vector<T> actual) {
     EXPECT_EQ(expected[i], actual[i]);
   }
 }
+
+// NB: This is not really sound, but all of the type sets constructed here
+// are singletons so it's fine
```

Comment:
should we assert this?

Reply:
Since this is test code I'm not too concerned about it.

Reply:
ya, I meant `legacyExtractTypeId` more generally -- is that sound?

Reply:
Yeah, `legacyExtractTypeId` is sound. The unsoundness that I'm remarking about here, is that arguably we SHOULD be testing against the entirety of the type set, not just a single type id. But since there's no variables in the c10 tests, it doesn't matter.